src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Services/UkrlpService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using SFA.DAS.RoatpCourseManagement.Application.UkrlpData;
+using SFA.DAS.RoatpCourseManagement.Configuration;
+using SFA.DAS.RoatpCourseManagement.InnerApi.Models.Ukrlp;
+
+namespace SFA.DAS.RoatpCourseManagement.Services
+{
+    public class UkrlpService: IUkrlpService
+    {
+        private readonly IUkrlpSoapSerializer _serializer;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<UkrlpService> _logger; 
+        private readonly UkrlpApiConfiguration _ukrlpConfiguration;
+
+        public UkrlpService(HttpClient httpClient, ILogger<UkrlpService> logger, IUkrlpSoapSerializer serializer, IOptions<UkrlpApiConfiguration> options)
```

Comment:
Why use a service instead of handler?

Reply:
I used the pattern that was used in coursedirectory calls, where the service goes and collects configuration, uses its own client for calls etc.  So it seemed the best match for the similar ukrlp service calls.

Reply:
I am not convinced. I would rather keep our solution consistent to a strategy instead of mixing things up and make is confusing for someone else to decide which way to go.

Reply:
Handler can have exact same code without any changes, if that is what you want to do.

Reply:
I've done that but the new code that was in 258 is causing sonarcloud to tip into saying the complexity is too high.  I've finesses it out now though.